File: AgentieTursimCS/Domain/Angajat.cs
using System;

namespace AgentieTurismCS.Domain
{
    public class Angajat : Entity<Guid>
    {
        private string _nume;
        private string _prenume;
        private string _username;
        private string _password;
        private string _agentieTurism;

        public Angajat(string nume, string prenume, string username, string password, string agentieTurism)
        {
            Id = Guid.NewGuid();
            _nume = nume;
            _prenume = prenume;
            _username = username;
            _password = password;
            _agentieTurism = agentieTurism;
        }

        public string Nume
        {
            get => _nume;
            set => _nume = value;
        }

        public string Prenume
        {
            get => _prenume;
            set => _prenume = value;
        }

        public string Username
        {
            get => _username;
            set => _username = value;
        }

        public string Password
        {
            get => _password;
            set => _password = value;
        }

        public string AgentieTurism
        {
            get => _agentieTurism;
            set => _agentieTurism = value;
        }

        public override bool Equals(object? obj)
        {
            if (this == obj) return true;
            if (!(obj is Angajat angajat)) return false;
            return _nume == angajat._nume &&
                   _prenume == angajat._prenume &&
                   _username == angajat._username &&
                   _password == angajat._password &&
                   _agentieTurism == angajat._agentieTurism;
        }
        
        public override int GetHashCode()
        {
            return HashCode.Combine(_nume, _prenume, _username, _password, _agentieTurism);
        }



        public override string ToString()
        {
            return
                $"Angajat{{nume='{_nume}', prenume='{_prenume}', username='{_username}', password='{_password}', agentieTurism='{_agentieTurism}'}}";
        }
    }
}

File: AgentieTurismJava/src/main/java/org/example/agentietursimjava/Domain/Angajat.java
package org.example.agentietursimjava.Domain;


import java.util.Objects;
import java.util.UUID;

public class Angajat extends Entity<UUID>{
    private String nume;
    private String prenume;
    private String username;
    private String password;
    private String agentieTurism;

    public Angajat(String nume, String prenume, String username, String password, String agentieTurism) {
        this.setId(UUID.randomUUID());
        this.nume = nume;
        this.prenume = prenume;
        this.username = username;
        this.password = password;
        this.agentieTurism = agentieTurism;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getAgentieTurism() {
        return agentieTurism;
    }

    public void setAgentieTurism(String agentieTurism) {
        this.agentieTurism = agentieTurism;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Angajat angajat)) return false;
        return Objects.equals(nume, angajat.nume) && Objects.equals(prenume, angajat.prenume) && Objects.equals(username, angajat.username) && Objects.equals(password, angajat.password) && Objects.equals(agentieTurism, angajat.agentieTurism);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nume, prenume, username, password, agentieTurism);
    }

    @Override
    public String toString() {
        return "Angajat{" +
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", username='" + username + '\'' +
                ", password='" + password + '\'' +
                ", agentieTurism='" + agentieTurism + '\'' +
                '}';
    }
}

